### PR TITLE
docs: deploy documentation site

### PR DESCRIPTION
**Pull request by ashleythedeveloper**
## What type of PR is this? (check all applicable)
- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [x] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [x] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR adds a GitHub Actions workflow to deploy the documentation site to GitHub Pages. It also updates various documentation files to include a disclaimer about the content being under development. The PR includes configuration changes for the documentation site deployment and minor updates to the documentation structure.

## Mobile & Desktop Screenshots/Recordings
Deployed site - https://uncefact.github.io/project-vckit/

Screenshot 2024-08-14 at 8 34 41 PM

## Added tests?
- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?
- [ ] 📜 README.md
- [x] 📓 [vc-kit doc site](https://uncefact.github.io/vckit/)
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: development/did-web-identifier [hunk 1]
```diff
@@ -0,0 +1 @@
+{"did":"did:web:hoa-ngo-gs.github.io:test-did","alias":"hoa-ngo-gs.github.io:test-did","provider":"did:web","controllerKeyId":"24674f77e846bb91e5dd38284856e435c93275550b9f7adbdc2f6d6df941afbc","keys":[{"kid":"24674f77e846bb91e5dd38284856e435c93275550b9f7adbdc2f6d6df941afbc","kms":"local","type":"Ed25519","publicKeyHex":"24674f77e846bb91e5dd38284856e435c93275550b9f7adbdc2f6d6df941afbc","privateKeyHex":"2249daba77ec537c67a677729cc92698bebd94e0d6f693bcfaedd22b689affc6e475b926b05b8ac5b66fab52a9d11b890a05321cab4842279a9bf83c61c365bd91f3714ea488a744ab0ee3ee1d6f1161f553575ba1de2ba70fcafe94685a5734c2e14988de656ee1197463e3f3d3a76866fe73f2286ac022d6887c8cc7c4a01650edc9b2646dba4f1eea7af6030b7a9237e28fe83da416bace3626c39f750075e29c69a247d16813","meta":{"algorithms":["EdDSA","Ed25519"]}}]}
```
Comment (ldhyen99) [inline development/did-web-identifier:1]: Should we change the did for the generic name?

File: development/did.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+    "did": "did:web:did:web:hoa-ngo-gs.github.io:test-did",
```
Comment (ldhyen99) [inline development/did.json:2]: Should we change the did for the generic name?

File: .github/workflows/buld_publish_docs.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+name: Deploy to GitHub Pages
```
Comment (huynguyen-hl) [inline .github/workflows/buld_publish_docs.yml:1]: I believe the workflow file should be named `.github/workflows/build_publish_docs.yml`.

Comment (ashleythedeveloper) [reply .github/workflows/buld_publish_docs.yml]: Good catch @huynguyen-hl. The file name has been updated.

**Review by hoa-ngo-gs — APPROVED** [review]

**Review by huynguyen-hl — APPROVED** [review]

**Pull request merged**